extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesUtil.java

```diff
@@ -0,0 +1,301 @@
+                BytecodeCreator hasNextFalse = hasNextBranchResult.falseBranch();
+
+                ResultHandle token = hasNextTrue.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(Iterator.class, "next", Object.class),
+                        iterator);
+                ResultHandle convertedObject = hasNextTrue.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(SmallRyeConfig.class, "convert", Object.class, String.class, Class.class),
+                        smallryeConfig, token, classToConvertTo);
+                hasNextTrue.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(Collection.class, "add", boolean.class, Object.class),
+                        collection, convertedObject);
+
+                hasNextTrue.continueScope(loop);
+                hasNextFalse.breakScope(loop);
+
+                methodCreator.returnValue(null);
+            }
+        }
+        return generatedClassName;
+    }
+
+    /**
+     * Generates code that uses config.getValue value to obtain the value of the property.
+     * When the result type is a Collection, the string value is obtained from MP config, then
+     * split and each token is converted into the required type
+     * If the the result type is not a collection, no conversion is performed
+     *
+     * @param generateConfigPropertiesUtilClassName The name of the generated utility class used to convert values if needed
+     * @param propertyName Property name that needs to be fetched
+     * @param resultType Type to which the property value needs to be converted to
+     * @param declaringClass Config class where the type was encountered
+     * @param bytecodeCreator Where the bytecode will be generated
+     * @param config Reference to the MP config object
+     */
+    static ResultHandle createReadMandatoryValueAndConvertIfNeeded(String generateConfigPropertiesUtilClassName,
+            String propertyName,
+            Type resultType,
+            DotName declaringClass,
+            BytecodeCreator bytecodeCreator, ResultHandle config) {
+        EffectiveTypeResponse effectiveTypeResponse = getEffectiveResultType(resultType, declaringClass);
+
+        ResultHandle value = bytecodeCreator.invokeInterfaceMethod(
+                MethodDescriptor.ofMethod(Config.class, "getValue", Object.class, String.class, Class.class),
+                config, bytecodeCreator.load(propertyName),
+                bytecodeCreator.loadClass(effectiveTypeResponse.getEffectiveType().toString()));
+
+        return createConversionToFinalValue(generateConfigPropertiesUtilClassName, resultType.name(),
+                effectiveTypeResponse.getEffectiveType(),
+                effectiveTypeResponse.getGenericType(), bytecodeCreator, value, config);
+    }
+
+    /**
+     * Generates code that uses config.getOptional value to obtain the value of the property.
```

Comment:
See also [`newOptionalConverter`](https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/Converters.java#L235-L245)

Reply:
Actually this doesn't appear to be in version `1.3.9` of Smallrye Config. How about we leave it as is for now and then I'll update the code when a new version lands?

Reply:
Maybe we should create an issue? Pretty sure we will forget about it.

Reply:
I was planning on it yes once the PR is merged

Reply:
Not a problem at all! I'll remove it when the new version lands

Reply:
Opened #4166 to clean this up when Smallrye Config is released